Panic during AutoEvent - reading Users

> **NOTE:** This issue happens randomly, and is not always reproduce-able.

**Provision Watcher**
```json
{
  "provisionwatcher":{
     "apiVersion":"v2",
     "name":"Test-Provision-Watcher_Emulator",
     "adminState":"UNLOCKED",
     "identifiers":{
        "Manufacturer": "Happytimesoft",
        "Model": "IPCamera"
     },
     "serviceName": "device-onvif-camera",
     "profileName": "onvif-camera",
     "autoEvents": [
         { "interval": "15s", "sourceName": "Users" }
      ]
  },
  "apiVersion":"v2"
}
```

**Panic**
```
level=DEBUG ts=2022-02-17T17:23:55.687729987Z app=device-onvif-camera source=discovery.go:35 msg="protocol discovery triggered"
Onvif WS-Discovery: Find 192.168.0.215:2020 
Onvif WS-Discovery: Find 192.168.0.53:5357 
Error 192.168.0.53:5357
camera is not available at 192.168.0.53:5357 or it does not support ONVIF services
level=DEBUG ts=2022-02-17T17:23:56.759283954Z app=device-onvif-camera source=onvifclient.go:227 msg="SOAP Request: <tds:GetDeviceInformation></tds:GetDeviceInformation>"
level=DEBUG ts=2022-02-17T17:23:56.789358733Z app=device-onvif-camera source=onvifclient.go:245 msg="SOAP Response: <GetDeviceInformationResponse><Manufacturer>TP-Link</Manufacturer><Model>C200</Model><FirmwareVersion>1.1.16 Build 211209 Rel.37726n(4555)</FirmwareVersion><SerialNumber>00c20000</SerialNumber><HardwareId>1.0</HardwareId></GetDeviceInformationResponse>"
level=DEBUG ts=2022-02-17T17:23:56.789400759Z app=device-onvif-camera source=driver.go:333 msg="Discovered camera from the address '192.168.0.215:2020'"
level=DEBUG ts=2022-02-17T17:23:56.789450723Z app=device-onvif-camera source=async.go:140 msg="Device TP-Link-C200-00c20000's Manufacturer value TP-Link did not match PW identifier: Happytimesoft"
level=DEBUG ts=2022-02-17T17:23:56.789476407Z app=device-onvif-camera source=async.go:127 msg="Filtered device addition finished"
level=DEBUG ts=2022-02-17T17:24:10.68887651Z app=device-onvif-camera source=executor.go:52 msg="AutoEvent - reading Users"
level=DEBUG ts=2022-02-17T17:24:10.688932544Z app=device-onvif-camera source=command.go:127 msg="Application - readDeviceResource: reading deviceResource: Users; X-Correlation-ID: "
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x8efc0e]
goroutine 61 [running]:
github.com/IOTechSystems/device-onvif-camera/internal/driver.(*DeviceClient).callOnvifFunction(0x0, 0xc00058b560, 0x6, 0xc00058b568, 0x8, 0xdf8c70, 0x0, 0x0, 0x0, 0x0, ...)
    /device-onvif-camera/internal/driver/onvifclient.go:217 +0x14e
github.com/IOTechSystems/device-onvif-camera/internal/driver.(*DeviceClient).CallOnvifFunction(0x0, 0xc00011c8f7, 0x5, 0xc00009bb30, 0xc00011c910, 0x6, 0xa36b4c, 0xb, 0xdf8c70, 0x0, ...)
    /device-onvif-camera/internal/driver/onvifclient.go:143 +0x14b
github.com/IOTechSystems/device-onvif-camera/internal/driver.(*Driver).HandleReadCommands(0xc0003dec00, 0xc000028ac0, 0x1d, 0xc0005122d0, 0xc00009af00, 0x1, 0x1, 0x1, 0x5, 0x0, ...)
    /device-onvif-camera/internal/driver/driver.go:146 +0x1ba
github.com/edgexfoundry/device-sdk-go/v2/internal/application.(*CommandProcessor).ReadDeviceResource(0xc00053bba8, 0xc0005159e0, 0xc, 0xc0005159db)
    /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.1.0/internal/application/command.go:146 +0x55b
github.com/edgexfoundry/device-sdk-go/v2/internal/application.CommandHandler(0x970001, 0x0, 0x0, 0xc0005cdd78, 0x0, 0x0, 0x0, 0xc0003d5620, 0x0, 0x0, ...)
    /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.1.0/internal/application/command.go:107 +0x59d
github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.readResource(0xc0000c2190, 0xc0003d5620, 0x16, 0xc0002984f0, 0x1)
    /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.1.0/internal/autoevent/executor.go:89 +0x1a6
github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.(*Executor).Run(0xc0000c2190, 0xae8c58, 0xc0003dec40, 0xc000416110, 0xc0000c6070, 0xc0003d5620)
    /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.1.0/internal/autoevent/executor.go:53 +0x296
created by github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.(*manager).triggerExecutors
    /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.1.0/internal/autoevent/manager.go:78 +0x245
```

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,7 @@
 module github.com/edgexfoundry/device-onvif-camera
 
 require (
-	github.com/IOTechSystems/onvif v0.0.2-0.20220103021116-4543eed40749
+	github.com/IOTechSystems/onvif v0.0.2-0.20220223143907-b50211157c1b
 	github.com/edgexfoundry/device-sdk-go/v2 v2.1.0
 	github.com/edgexfoundry/go-mod-bootstrap/v2 v2.1.0
 	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.1.0

--- a/internal/driver/driver.go
+++ b/internal/driver/driver.go
@@ -103,19 +103,19 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.As
 func (d *Driver) getDeviceClient(deviceName string) (*DeviceClient, errors.EdgeX) {
 	d.lock.RLock()
 	defer d.lock.RUnlock()
-	c, ok := d.deviceClients[deviceName]
+	deviceClient, ok := d.deviceClients[deviceName]
 	if !ok {
 		device, err := sdk.RunningService().GetDeviceByName(deviceName)
 		if err != nil {
 			return nil, errors.NewCommonEdgeXWrapper(err)
 		}
-		deviceClient, err := NewDeviceClient(device, d.config, d.lc)
+		deviceClient, err = NewDeviceClient(device, d.config, d.lc)
 		if err != nil {
 			return nil, errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("failed to initial device client for '%s' camera", device.Name), err)
 		}
 		d.deviceClients[deviceName] = deviceClient
 	}
-	return c, nil
+	return deviceClient, nil
 }
 
 func (d *Driver) removeDeviceClient(deviceName string) {

